src/core/utils/components/InputContainer.tsx

```diff
@@ -49,7 +49,7 @@ export const InputContainer = <T extends React.ElementType = 'div'>(
           [`iui-${status}`]: !!status,
           'iui-inline-label': isLabelInline,
           'iui-inline-icon': isIconInline,
-          'iui-with-message': !!message && !isLabelInline,
+          'iui-with-message': (!!message || !!icon) && !isLabelInline,
```

Comment:
@veekeys any comments/concerns here?

Reply:
Just was trying to think about situation with having no message, but having icon with `isIconInline` option. Would it have some issues?

Reply:
Why is this change needed?

Reply:
https://github.com/iTwin/iTwinUI-react/pull/554#discussion_r810958103